app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/ExposureNotificationsRepository.kt

```diff
@@ -259,14 +275,46 @@ class ExposureNotificationsRepository @Inject constructor(
     }
 
     suspend fun checkExposure(context: Context) {
-        val lastExposure = getLastRiskyExposure()?.daysSinceEpoch
-        val lastNotifiedExposure = prefs.getLastNotifiedExposure()
-        if (lastExposure != null && lastExposure != lastNotifiedExposure) {
+        importLegacyExopsures()
+        db.dao().deleteOld()
+        db.dao().insert(getDailySummariesFromApi(true).map {
+            DailySummaryEntity(
+                daysSinceEpoch = it.daysSinceEpoch,
+                maximumScore = it.summaryData.maximumScore,
+                scoreSum = it.summaryData.scoreSum,
+                weightenedDurationSum = it.summaryData.weightedDurationSum,
+                importTimestamp = System.currentTimeMillis(),
+                notified = false,
+                accepted = false
+            )
+        })
+        val latestExposure = db.dao().getLatest().firstOrNull()?.daysSinceEpoch
+        val lastNotifiedExposure = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
+        if (latestExposure != null && latestExposure != lastNotifiedExposure) {
             LocalNotificationsHelper.showRiskyExposureNotification(context)
-            prefs.setLastNotifiedExposure(lastExposure)
+            db.dao().markAsNotified()
             firebaseFunctionsRepository.registerNotification()
         } else {
-            L.i("Not showing notification, lastExposure=$lastExposure, lastNotifiedExposure=$lastNotifiedExposure")
+            L.i("Not showing notification, lastExposure=$latestExposure, lastNotifiedExposure=$lastNotifiedExposure")
+        }
+    }
+
+    //TODO: Remove in late november 2020
+    private suspend fun importLegacyExopsures(){
+        if (!prefs.isLegacyExposuresImported()){
+            db.dao().insert(getDailySummariesFromApi(false).map {
+                DailySummaryEntity(
+                    daysSinceEpoch = it.daysSinceEpoch,
+                    maximumScore = it.summaryData.maximumScore,
+                    scoreSum = it.summaryData.scoreSum,
+                    weightenedDurationSum = it.summaryData.weightedDurationSum,
+                    importTimestamp = if (it.daysSinceEpoch > prefs.getLastNotifiedExposure()) System.currentTimeMillis() else 0,
```

Comment:
Nechápu tenhle if, proč tam je?

Reply:
abychom stará setkání mohli v nadcházejícím groupování podle dne kdy byly identifikované, jednoznačně rozlišit a dát je to sekce "dřívější setkání"